app/root.tsx

```diff
@@ -307,6 +307,23 @@ function PageLoadingMessage() {
   )
 }
 
+declare global {
+  const fathom: {
+    trackPageview(): void
+  }
+}
+
+function CanonicalLink({origin}: {origin: string}) {
+  const {pathname} = useLocation()
+  const canonicalUrl = removeTrailingSlash(`${origin}${pathname}`)
+
+  React.useEffect(() => {
+    fathom.trackPageview()
```

Comment:
Is it possible this could run before fathom is finished loading? Should we update the types/code to reflect this possibility?

Reply:
I don't know, that's a good question… Maybe it would be better to sacrifice the console and use `fathom-client` instead, since it's specifically designed to be able to track before the script has finished loading.

I updated the code.